src/mattack_actors.cpp

```diff
@@ -153,7 +153,12 @@ void mon_spellcasting_actor::load_internal( JsonObject &obj, const std::string &
     mandatory( obj, was_loaded, "spell_id", sp_id );
     optional( obj, was_loaded, "self", self, false );
     optional( obj, was_loaded, "spell_level", spell_level, 0 );
-    spell_data = spell( spell_id( sp_id ) );
+    // ~ translator "<Monster Display name> cast <Spell Name>!"
+    std::string default_msg = "%s cast %s!";
+    std::string monster_message;
+    optional( obj, was_loaded, "monster_message", monster_message, default_msg );
+    translation msg = translation( monster_message );
+    spell_data = spell( spell_id( sp_id ), msg );
```

Comment:
You can read `translation` directly from json to make it support translation context. For example:

```
translation monster_message;
optional( obj, was_loaded, "monster_message", monster_message, translation( "%s cast %s!" ) );
spell_data = spell( spell_id( sp_id ), monster_message );
```

Might also need to update `lang/extract_json_strings.py` to extract "monster_message" strings for translation.

Reply:
I believe I attempted this version first and got an error that optional is for strings?

Reply:
Well, `optional` has a templated version, not sure why it did not work... Could you post the exact error message?

Reply:
ah nm, got it to work now. thanks